trunner/test_runner.py

```diff
@@ -191,29 +194,35 @@ def run_tests(self, tests: Sequence[TestOptions]):
             def set_reboot_flag(tests, idx, result):
                 # If the test is successful and the next test doesn't require loading via
                 # the plo we don't want to reboot the entire device (to speed up the test execution).
-                # There are three exceptions to this rule:
+                # There are five exceptions to this rule:
+                # 0. The yaml config has been configured with the reboot parameter set to True.
                 # 1. Runner runs in the "nightly" mode when we are not concerned about slow execution.
                 # 2. The test has failed.
                 # 3. We have to enter the bootloader in order to load applications.
+                # 4. The test is skipped, but there should be a reboot.
+                nonlocal next_reboot
+
                 if idx == len(tests) - 1:
                     return
 
-                tests[idx + 1].should_reboot = False
-
-                if result.is_skip():
-                    tests[idx + 1].should_reboot = tests[idx].should_reboot
-
                 if (
-                    result.is_fail()
-                    or self.ctx.nightly
+                    self.ctx.nightly
                     or (
                         not self.ctx.target.rootfs
                         and tests[idx + 1].bootloader is not None
                         and tests[idx + 1].bootloader.apps
                     )
+                    or result.is_skip() and next_reboot
                 ):
                     tests[idx + 1].should_reboot = True
 
+                if not result.is_skip():
```

Comment:
I know that it's not the most optimal solution to set it to False each time, but I hope it's not a big problem. Alternatively:

```suggestion
                if not result.is_skip() and next_reboot:
```